bodhi-skopeo-lite/Dockerfile

```diff
@@ -0,0 +1,8 @@
+FROM fedora:29
+
+RUN dnf -y --refresh install \
+        /usr/bin/bodhi-skopeo-lite \
```

Comment:
`/usr/bin/bodhi-skopeo-lite` is not as lite as you could assume.

```
[root@bfaf00824636 /]# dnf install -d1 /usr/bin/skopeo
Last metadata expiration check: 0:02:17 ago on Mon 03 Dec 2018 05:02:20 PM UTC.
Dependencies resolved.
================================================================================
 Package             Arch     Version                           Repository
                                                                           Size
================================================================================
Installing:
 skopeo              x86_64   1:0.1.32-2.dev.gite814f96.fc29    fedora    7.9 M
Installing dependencies:
 ostree-libs         x86_64   2018.9-1.fc29                     updates   361 k
 containers-common   x86_64   1:0.1.32-2.dev.gite814f96.fc29    fedora     31 k

Transaction Summary
================================================================================
Install  3 Packages

Total download size: 8.3 M
Installed size: 24 M
```

vs
```
[root@bfaf00824636 /]# dnf install -d1 /usr/bin/bodhi-skopeo-lite
Last metadata expiration check: 0:03:13 ago on Mon 03 Dec 2018 05:02:20 PM UTC.
Dependencies resolved.
================================================================================
 Package                               Arch   Version             Repository
                                                                           Size
================================================================================
Installing:
 bodhi-server                          noarch 3.11.1-1.fc29       updates 5.3 M
Installing dependencies:
 Lmod                                  x86_64 7.8.9-1.fc29        updates 218 k
 cairo                                 x86_64 1.16.0-1.fc29       updates 676 k
 cups-libs                             x86_64 1:2.2.8-6.fc29      updates 323 k

//snip

 xorg-x11-fonts-ISO8859-1-100dpi       noarch 7.5-20.fc29         fedora  1.1 M
 xorg-x11-server-utils                 x86_64 7.7-26.fc29         fedora  192 k
 xz                                    x86_64 5.2.4-3.fc29        fedora  146 k
 zeromq                                x86_64 4.1.6-7.fc28        fedora  248 k
Installing weak dependencies:
 graphviz                              x86_64 2.40.1-37.fc29      updates 1.7 M
 python-unversioned-command            noarch 2.7.15-11.fc29      updates  13 k
 python2-pip                           noarch 18.1-1.fc29         updates 1.9 M
 python2-unbound                       x86_64 1.8.1-1.fc29        updates  94 k
 ImageMagick                           x86_64 1:6.9.9.38-3.fc29   fedora  163 k
 apr-util-bdb                          x86_64 1.6.1-8.fc29        fedora   12 k
 apr-util-openssl                      x86_64 1.6.1-8.fc29        fedora   14 k
 perl-IO-Socket-IP                     noarch 0.39-418.fc29       fedora   42 k
 perl-IO-Socket-SSL                    noarch 2.060-2.fc29        fedora  237 k
 perl-Mozilla-CA                       noarch 20180117-3.fc29     fedora   12 k
 python2-beaker                        noarch 1.10.0-1.fc29       fedora  108 k
 python2-crypto                        x86_64 2.6.1-24.fc29       fedora  477 k
 python2-cssselect                     noarch 0.9.2-9.fc29        fedora   38 k

Transaction Summary
================================================================================
Install  370 Packages

Total download size: 144 M
Installed size: 533 M
```

And even with disabled week dependencies it is much more that full version of skopeo
```
[root@bfaf00824636 /]# dnf install -d1 --setopt=install_weak_deps=False /usr/bin/bodhi-skopeo-lite
Last metadata expiration check: 0:04:43 ago on Mon 03 Dec 2018 05:02:20 PM UTC.
Dependencies resolved.
================================================================================
 Package                               Arch   Version             Repository
                                                                           Size
================================================================================
Installing:
 bodhi-server                          noarch 3.11.1-1.fc29       updates 5.3 M
Installing dependencies:
 Lmod                                  x86_64 7.8.9-1.fc29        updates 218 k
 freetype                              x86_64 2.9.1-6.fc29        updates 357 k
 gettext                               x86_64 0.19.8.1-18.fc29    updates 1.0 M
 gettext-common-devel                  noarch 0.19.8.1-18.fc29    updates 397 k

//snip

 python3-pyOpenSSL                     noarch 18.0.0-3.fc29       fedora   95 k
 python3-pycparser                     noarch 2.14-15.fc29        fedora  146 k
 python3-pycurl                        x86_64 7.43.0.2-3.fc29     fedora  220 k
 python3-pysocks                       noarch 1.6.8-4.fc29        fedora   33 k
 python3-requests-kerberos             noarch 0.12.0-4.fc29       fedora   26 k
 tcl                                   x86_64 1:8.6.8-2.fc29      fedora  1.1 M
 xz                                    x86_64 5.2.4-3.fc29        fedora  146 k
 zeromq                                x86_64 4.1.6-7.fc28        fedora  248 k

Transaction Summary
================================================================================
Install  268 Packages

Total download size: 107 M
Installed size: 397 M
```

Reply:
That is kind of a large thing to install though... Is there a way we can install just the skopeo-lite portion of it?
The [file](https://github.com/fedora-infra/bodhi/blob/develop/bodhi/server/scripts/skopeo_lite.py) itself seems to not have any dependency in Bodhi. Maybe just download it directly from source?

Reply:
> @lslebodn AFAICT, `bodhi-skopeo-lite` is less about the 'lite' and more about the 'bodhi' -- it [supports manifest lists](https://github.com/fedora-infra/bodhi/commit/a0496fc9d7c3a69ac3e2519b8d560d0df0fb449f); "regular" skopeo doesn't ([yet](https://github.com/containers/image/pull/400))

It is not really true that skopeo does not support.
```
sh$ skopeo inspect --raw docker://docker.io/fedora:rawhide
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 529,
         "digest": "sha256:24f78d0b989b20b4aff0ce14f064884957600ae721f281947d28637d703ac75d",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 529,
         "digest": "sha256:c95d84e6bf7681187ed820c9ab67508887aa3f6d7a8f57d36f89309ccdc2a094",
         "platform": {
            "architecture": "arm64",
            "os": "linux",
            "variant": "v8"
         }
      }
   ]
}
``` 

But you probably meant copy of `manifest v2 list` :-)



> That is kind of a large thing to install though... Is there a way we can install just the skopeo-lite portion of it?
> The [file](https://github.com/fedora-infra/bodhi/blob/develop/bodhi/server/scripts/skopeo_lite.py) itself seems to not have any dependency in Bodhi. Maybe just download it directly from source?

The size was my main point. BTW on rawhide bodhi-server does not depend on python2 but it is still 83 MB
because of simple python script.
```
[root@44f378f0ae6b /]# dnf install /usr/bin/bodhi-skopeo-lite --setopt=install_weak_deps=False
Fedora - Modular Rawhide - Developmental packag 388 kB/s | 2.0 MB     00:05    
Fedora - Rawhide - Developmental packages for t 1.9 MB/s |  63 MB     00:33    
Last metadata expiration check: 0:00:01 ago on Mon 03 Dec 2018 08:04:27 PM UTC.
Dependencies resolved.
================================================================================
 Package                     Arch   Version                       Repository
                                                                           Size
================================================================================
Installing:
 bodhi-server                noarch 3.11.1-100.fc30               rawhide 5.3 M
Installing dependencies:
 apr                         x86_64 1.6.5-1.fc30                  rawhide 106 k

//snip

 python3-zope-schema         noarch 4.4.2-12.fc29                 rawhide 229 k
 xz                          x86_64 5.2.4-4.fc30                  rawhide 143 k
 zeromq                      x86_64 4.1.6-10.fc30                 rawhide 226 k

Transaction Summary
================================================================================
Install  192 Packages
Upgrade    6 Packages

Total download size: 83 M
```

Reply:
Yes I know that image is quite big because all the bodhi stuff. However I wanted to keep it robust in case that code/dependencies change.

I will rework this to just py script + dependencies

Reply:
reworked